### PR TITLE
Added support for the node.roles configuration.

### PR DESCRIPTION
**Pull request by mhjtrifork**
**Description of the change**

Added support for the node.roles configuration instead of the deprecated node.master and node.data options. The change should be compatible with existing deployments but setting the ELASTICEARCH_USE_NODE_ROLES option will ignore any ELASTICSEARCH_NODE_TYPE settings.

**Benefits**

Allows setting of all the specific node roles, enabling functionality such as ILM by using roles such as data_hot and data_warm.

**Possible drawbacks**

None

**Applicable issues**

N/A

**Additional information**

Only changed for ES 7 image.

### CODE REVIEW

File: 7/debian-10/rootfs/opt/bitnami/scripts/libelasticsearch.sh [hunk 1]
```diff
@@ -469,9 +539,12 @@ elasticsearch_initialize() {
...
         is_boolean_yes "$ELASTICSEARCH_LOCK_ALL_MEMORY" && elasticsearch_conf_set bootstrap.memory_lock "true"
         elasticsearch_cluster_configuration
-        elasticsearch_configure_node_type
+        if is_boolean_yes "$ELASTICSEARCH_USE_NODE_ROLES"; then
+            elasticsearch_configure_node_roles
+        else
+            elasticsearch_configure_node_type
+        fi
         elasticsearch_custom_configuration
-        [[ -d "${ELASTICSEARCH_BASE_DIR}/modules/x-pack-ml/platform/linux-x86_64/lib" ]] || elasticsearch_conf_set xpack.ml.enabled "false"
```
Comment (Mauraza) [inline 7/debian-10/rootfs/opt/bitnami/scripts/libelasticsearch.sh:474]: Could you add this line?
it's necessary

**Review by Mauraza — CHANGES_REQUESTED** [review]
Hi @mhjtrifork,

Thanks for the PR! Could you check the comment?

File: 7/debian-10/rootfs/opt/bitnami/scripts/libelasticsearch.sh (hunk 1 above)
Comment (mhjtrifork) [reply 7/debian-10/rootfs/opt/bitnami/scripts/libelasticsearch.sh:474]: Ups, missed that one. Added again.

**Comment by Mauraza** [comment]
Hi @mhjtrifork,

I going to do some internal tests to check that everything goes well. we will update the thread with the results.

**Comment by mhjtrifork** [comment]
> Hi @mhjtrifork,
> 
> I going to do some internal tests to check that everything goes well. we will update the thread with the results.

Great, thanks.

I will probably add a new PR today/tomorrow for supporting multiple data paths.

**Comment by Mauraza** [comment]
Hi @mhjtrifork,

I going to do the test with the new changes. Thanks for the PR!

**Comment by marcosbc** [comment]
Removing the `on-hold` in order to unblock this. @Mauraza could you check if this can be merged?

**Comment by Mauraza** [comment]
I'm doing some tests, I will update the thread when this finish

**Comment by Mauraza** [comment]
Hi @mhjtrifork,

I'm making some changes, we will merge this PR soon

**Comment by alemorcuq** [comment]
We expect to have this merged in the following days.

**Pull request merged**